datalad/core/local/tests/test_status.py

```diff
@@ -242,3 +244,34 @@ def test_subds_status(path):
         type="dataset",
         path=op.join(subds.path, "someotherds"),
         refds=subds.path)
+
+
+@with_tempfile
+def test_status_symlinked_dir_within_repo(path):
+    if not has_symlink_capability():
+        raise SkipTest("Can't create symlinks")
+    # <path>
+    # |-- bar -> <path>/foo
+    # `-- foo
+    #     `-- f -> ../.git/annex/objects/[...]
+    ds = Dataset(path).create()
+    foo = ds.pathobj / "foo"
+    foo.mkdir()
+    (foo / "f").write_text("content")
+    (ds.pathobj / "bar").symlink_to(foo, target_is_directory=True)
+    ds.save()
+    bar_f = (ds.pathobj / "bar" / "f")
+
+    def call():
+        return ds.status(path=[bar_f], annex="availability",
+                         on_failure="ignore", result_renderer=None)
+
+    if ds.repo.git_annex_version < "8.20200522":
+        assert_result_count(call(), 0)
+    else:
+        # As of 2a8fdfc7d (Display a warning message when asked to operate on a
+        # file inside a symlinked directory, 2020-05-11), git-annex will error.
```

Comment:
Oi... I is that if a directory within repository or somewhere up the hierarchy?

Reply:
> Oi... I is that if a directory within repository or somewhere up the hierarchy?

This test is specifically about the within directory case.  In this situation, git-annex now fails instead of silently doing nothing.  Either behavior is consistent with git, depending on which git command you look at :)

However, as mentioned in the commit message, "[git-annex] does so in cases like above, but also in cases where the symlinked directory is upstream of the top-level directory of the repository".  That description should be improved because it doesn't explicitly mention that this is only relevant if absolute paths are used.  Anyway, this is the change in behavior that is behind the `_DL_TMPDIR="/var/tmp/sym link"` failures.

I think this change in behavior is worth raising on git-annex's side because the git-annex change was motivated by git's behavior in the within-repo case, but it's more aggressive than git is by extending to the symlinked directories upstream of the repo.

I haven't gotten around to filing that yet, but I think, even if Joey agrees with adjusting the new behavior, we should still make the change here for at least these reasons:

  * It makes `status` work for the "linked upstream directory" case for git-annex >= 8.20200522, with no downside that I'm aware of.

  * `status` already passes these GitRepo.path-recoded paths to `diffstatus`.  It then takes the result from that call, which is keyed by the recoded paths, and passes it as `init` to `get_content_annexinfo`, while passing the original Dataset.repo-based paths as the `paths` argument.  When those two sets don't match, this happens to not cause any issues (because paths come out of `annex find{,ref}` as relative and are joined to the GitRepo.path), but it's still inconsistent and unnecessarily confusing.

  * The commands are run with GitRepo.path as the current directory.  In the upstream symlink case, it's confusing to pass in absolute paths that are instead based off of Dataset.path, especially when we already have the GitRepo.path-adjusted paths at hand because we pass them to `diffstatus`.